File: python/bubbleSort.py
import sys
import time

def entrada(vetor):

    try:
        while True:
            vetor.append(int(input())) 
    except EOFError:
        print("Arquivo lido com sucesso!")

def bubbleSort(vetor):
    for j in range(len(vetor), 1, -1):
        for i in range(1, j):
            if vetor[i] < vetor[i-1]:
                aux = vetor[i]
                vetor[i] = vetor[i-1]
                vetor[i-1] = aux


#recebe a primeira linha do arquivo
nome_arquivo = input()

vetor = []
vetorcp = []
entrada(vetor)

for c in range(3):
    vetorcp = vetor.copy()

    inicio = time.perf_counter()
    bubbleSort(vetorcp)
    fim = time.perf_counter()

    #print(vetor)
    
    arquivo = open('python/output/bubbleSort.txt', 'a')
    arquivo.write(f"{(fim - inicio) * 1000}\t")

arquivo.write("\n")
arquivo.close()

File: java/BubbleSort.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class BubbleSort {
    public static void main(String[] args) {
        String nomeArquivo = null; 
        String caminhoArquivoEntrada = System.getProperty("user.dir") + "/../inputs/int_decrescente/8_500000_10000000.txt";
        String caminhoArquivoSaida = System.getProperty("user.dir") + "/../output/int_Bubble_java";
        try {
            BufferedReader br = new BufferedReader(new FileReader(caminhoArquivoEntrada));
            List<Integer> vetor = new ArrayList<>();
            nomeArquivo = br.readLine();
            String line;

            while ((line = br.readLine()) != null) {
                vetor.add(Integer.parseInt(line));
            }

            br.close();

            long inicio = System.currentTimeMillis();
            bubbleSort(vetor);
            long fim = System.currentTimeMillis();

            //System.out.println(vetor);

            try (FileWriter fw = new FileWriter(caminhoArquivoSaida, true)) {
                fw.write(nomeArquivo + "\n\tAlgoritmo de ordenacao: BubbleSort\n\tTempo de execucao: " + (fim - inicio) + " ms\n\tQuantidade de numeros: " + vetor.size() + "\n\n");
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void bubbleSort(List<Integer> vetor) {
        int n = vetor.size();
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (vetor.get(j) > vetor.get(j + 1)) {
                    int temp = vetor.get(j);
                    vetor.set(j, vetor.get(j + 1));
                    vetor.set(j + 1, temp);
                }
            }
        }
    }
}
